src/gwt/tools/build-js-vector-aliases.R

```diff
@@ -0,0 +1,56 @@
+root <- rprojroot::find_rstudio_root_file()
+setwd(root)
+
+transformations <- list(
+   
+   list(
+      name = "Integer",
+      type = "int",
+      null = "0"
+   ),
+   
+   list(
+      name = "Number",
+      type = "double",
+      null = "0"
+   ),
+   
+   list(
+      name = "String",
+      type = "String",
+      null = "null"
+   ),
+   
+   list(
+      name = "Boolean",
+      type = "boolean",
+      null = "false"
+   )
+   
+)
+
+SOURCE <- "src/gwt/src/org/rstudio/core/client/JsVector.java"
+
+contents <- readLines(SOURCE)
+lapply(transformations, function(transformation) {
```

Comment:
GWT has code generation built-in, which would allow you to bake this into the existing compilation phase instead of requiring a manual re-run of the R script when you change the input file. I'll let you decide if doing this is worth it, but it isn't terribly difficult. See `CommandBundleGenerator` for an example:

https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/core/rebind/command/CommandBundleGenerator.java